core-rust/state-manager/src/store/in_memory.rs

```diff
@@ -473,6 +473,37 @@ impl TransactionIdentifierLoader for InMemoryStore {
     }
 }
 
+pub struct InMemoryProofIterator<'a> {
+    _state_version: StateVersion,
+    _store: &'a InMemoryStore,
+}
+
+impl<'a> InMemoryProofIterator<'a> {
+    fn new(from_state_version: StateVersion, store: &'a InMemoryStore) -> Self {
+        Self {
+            _state_version: from_state_version,
+            _store: store,
+        }
+    }
+}
+
+impl Iterator for InMemoryProofIterator<'_> {
+    type Item = LedgerProof;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        unimplemented!()
+    }
+}
+
+impl IterableProofStore for InMemoryStore {
+    fn get_proof_iter(
+        &self,
+        from_state_version: StateVersion,
+    ) -> Box<dyn Iterator<Item = LedgerProof> + '_> {
+        Box::new(InMemoryProofIterator::new(from_state_version, self))
```

Comment:
Hah, I changed course mid work and left it like this. `unimplemented!` right here instead of iterator is more appropiate.

What would it take to completely remove the `InMemoryStore`? We'll have to do it at some point anyway and better sooner rather than later.

Reply:
> What would it take to completely remove the InMemoryStore? 

I wonder why we even need it. For tests? Rocksdb seems lightweight enough...

Reply:
Yes, just for tests.

It is going to be cumbersome to keep it when we add snapshots.

Reply:
I am all in for removing it completely (just not in this PR).